### PR TITLE
Task/favorite groups

### PR DESCRIPTION
**Pull request by drospond**
Description: 
-Creates a "Groups" section in the user dashboard for creating and managing groups of favorites. Currently limited to favorite documents.

Ticket: https://jira.di2e.net/browse/UOT-110601

Testing Instructions:
- From "Groups" tab on the User Dashboard
     -Create groups from button top right
          -Name can't be repeated
-From "Documents" section in the favorites tab of the User Dashboard
     -Add favorites to a group from button top right
          -Groups should only be able to take 5 documents. Error shows is 5 exceeded
     -Removing a favorite should also remove the item from any groups
- From "Groups" tab on the User Dashboard
     -Remove documents from group with star in top right corner
     -View favorites' info on card once clicked
          -View favorite cummary by clicking on speech bubbly icon on the bottom
          -Open document in PDF viewer with "Open" button (doesn't work on local)
          -Open details page with "Details" button
     -Exporting groups from button in top right corner of groups. Currently uses a combination of the favorites' saved search texts
     -Delete groups from button in top right

### CODE REVIEW

File: frontend/src/components/user/GCUserDashboard.js [hunk 1]
```diff
@@ -707,6 +816,73 @@ const GCUserDashboard = (props) => {
...
+		setDocumentsToGroup([]);
+	}
+
+	const renderDocumentsToAdd = () => {
+		let groupFavorites;
+		if(selectedGroup.name) groupFavorites = userData.favorite_groups.find(group => group.group_name === selectedGroup.name)?.favorites;
+		return <div style={{overflow: 'scroll', maxHeight: 300, display: 'grid', gridTemplateColumns: 'repeat(2, 1fr)'}}>
+			{_.map(favoriteDocuments, (doc) => {
+				if(selectedGroup.name){
+					if(groupFavorites.includes(doc.favorite_id)) {
```
Comment (aaron42051) [inline frontend/src/components/user/GCUserDashboard.js:866]: It only happened once, but when deleting a group I made this line errored out from groupFavorites being undefined. I think if this is handled somehow the rest looks good.

**Review by aaron42051 — COMMENTED** [review]
Exporting a group seems to timeout in the backend (when I tried locally: 9ZJ4HRN)

File: frontend/src/components/user/GCUserDashboard.js (hunk 1 above)
Comment (drospond) [reply frontend/src/components/user/GCUserDashboard.js:866]: The exporting issue is tied to a bug that I have another PR out for to fix

Comment (drospond) [reply frontend/src/components/user/GCUserDashboard.js]: Got a fix in for groupFavorites being undefined

**Review by aaron42051 — APPROVED** [review]

**Review by ryanryanchan — APPROVED** [review]

**Pull request merged**